open-craft/edx-analytics-pipeline
# fix: updates the python 2.7 get-pip url

## Description

This fixes the following issue with the analytics pipeline for Ginkgo.

```
TASK: [install pip] *********************************************************** 
failed: [10.12.254.17] => {"changed": true, "cmd": ["/var/lib/analytics-tasks/automation/venv/bin/python", "get-pip.py"], "delta": "0:00:00.012953", "end": "2021-03-24 13:07:21.843188", "rc": 1, "start": "2021-03-24 13:07:21.830235"}
stderr: 
Hi there!

The URL you are using to fetch this script has changed, and this one will no
longer work. Please use get-pip.py from the following URL instead:

    https://bootstrap.pypa.io/pip/2.7/get-pip.py

Sorry if this change causes any inconvenience for you!

We don't have a good mechanism to make more gradual changes here, and this
renaming is a part of an effort to make it easier to us to update these
scripts, when there's a pip release. It's also essential for improving how we
handle the `get-pip.py` scripts, when pip drops support for a Python minor
version.

There are no more renames/URL changes planned, and we don't expect that a need
would arise to do this again in the near future.

Thanks for understanding!

- Pradyun, on behalf of the volunteers who maintain pip.

FATAL: all hosts have already failed -- aborting
```

## Supporting information

* **Jira Tickets**: SE-4221, SE-4097
* **Related Pull Requests**: #14 

## Testing instructions

Anyway, I tested this using the following builds:
* [`course-activity-workflow#92`](https://admin.edx-flatu.org:8080/view/Workflows/job/course-activity-workflow/92/)
* [`answer-distribution-workflow#594`](https://admin.edx-flatu.org:8080/view/Workflows/job/answer-distribution-workflow/594/)

To verify the builds, click "View as plain text" and verify the following:
* Search for `--repo https://github.com/open-craft/edx-analytics-pipeline --branch nizar/python27-pip-pipeline-fix` and make sure it is present in the text. It will verify that the custom branch is used, and thus the fix is applied
* Search for `TASK: [install pip]` and verify the task did not fail. It will verify that the previous failure happening no longer happens

## Deadline

ASAP
